[BUG] TestRpullImageThenRemove is Flaky

**Describe the bug**
TestRpullImageThenRemove is flaky when I run it locally. It's still finding fuse mounts after removing the image. It looks like check fuse mounts is racing against the containerd garbage collector. Once the image is removed the garbage collector will remove snapshots async so if that happens after the check, there will still be mounts.


**Steps To Reproduce**

```
$ GO_TEST_FLAGS='-run TestRpullImageThenRemove' make integration
...
    pull_test.go:226: incorrect number of remote snapshots; expected=0, actual=3
--- FAIL: TestRpullImageThenRemove (12.31s)
FAIL
FAIL	github.com/awslabs/soci-snapshotter/integration	12.462s
FAIL

```

**Expected behavior**
The test passes consistently


**Configuration (please complete the following information):**
 - OS: Debian 10
 - Snapshotter Version: soci-snapshotter-grpc 6137c69.m 6137c697a6382707397a248824fa5ec2086d7a2f.m
 - Containerd Version: 1.6.8


**Additional context**
Add any other context about the problem here.

--- a/integration/pull_test.go
+++ b/integration/pull_test.go
@@ -840,8 +840,8 @@ func TestRpullImageThenRemove(t *testing.T) {
 
 	checkFuseMounts(t, sh, len(sociIndex.Blobs))
 
-	sh.X("ctr", "image", "rm", regConfig.mirror(containerImage).ref)
-	sh.X("ctr", "image", "rm", dockerhub(containerImage).ref)
+	sh.X("ctr", "image", "rm", "--sync", regConfig.mirror(containerImage).ref)
+	sh.X("ctr", "image", "rm", "--sync", dockerhub(containerImage).ref)
 
 	checkFuseMounts(t, sh, 0)
 }

